hera_cal/vis_clean.py

```diff
@@ -7,29 +7,240 @@
+    return wout
+
+
+def flag_rows_with_contiguous_flags(weights_in, max_contiguous_flag, ax='freq'):
+    """
+    flag any row or column with contiguous zero-weights over limit.
+
+    Parameters
+    ----------
+    weights_in : array-like, 2d
+        weights to check. any row (ax='time') or col (ax='freq')
+        with contiguous regions of zero with length greater then max_contiguous_flag
+        will be set to zero.
+    max_contiguous_flag : integer (or 2-list/tuple)
+        any row or column with contiguous weights equal to zero greater or equal to this value
+        will be set entirely to zero.
+    ax : str, optional
+        axis to perform flagging over.
+        valid options include 'time' (zeroth axis), 'freq'(1 axis)
+        or 'both'.
+        default is 'freq'
+    """
+    if ax == 'time':
+        wout = flag_rows_with_contiguous_flags(weights_in.T, max_contiguous_flag).T
+    else:
+        wout = copy.deepcopy(weights_in)
+        for rownum, wrow in enumerate(wout):
+            max_contiguous = 0  # keeps track of the largest contig flags in integration.
+            current_flag_length = 0  # keeps track of current contig flag size.
+            on_flag = False  # keep track if currently on a flag.
+            # iterate over each channel in integration.
+            for wr in wrow:
+                on_flag = (wr == 0)  # if weights are zero, on a flag.
+                # if on a flag, +1 current flag size.
+                if on_flag:
+                    current_flag_length += 1
+                else:
+                    # otherwise, check if the current flag len > max.
+                    # update max_contiguous if appropriate.
+                    if current_flag_length >= max_contiguous:
+                        max_contiguous = current_flag_length
+                    current_flag_length = 0
+            if ax == 'both':
+                if max_contiguous >= max_contiguous_flag[1]:
+                    wout[rownum][:] = 0.
+            elif max_contiguous >= max_contiguous_flag:
+                wout[rownum][:] = 0.
+        if ax == 'both':
+            wout = flag_rows_with_contiguous_flags(wout, max_contiguous_flag[0], ax='time')
+    return wout
+
+
+def get_max_contiguous_flag_from_filter_periods(x, filter_centers, filter_half_widths):
+    """
+    determine maximum contiguous flags from filter periods
+
+    Parameters
+    ----------
+    x : array-like, 1d or 2-tuple
+        x (and y) axes of data to determine maximum contiguous flags from.
+    filter_centers : list or 2-tuple/list of lists
+        centers of filtering-windows.
+    filter_half_widths : list or 2-tuple list of lists
+        half-widths of filtering windows.
+
+    Returns
+    -------
+    max_contiguous_flag: int or 2-list containing the width of a region corresponding
+                         to the largest delay in the filter centers and filter_widths
```

Comment:
indentation: btw I noticed that you tend to indent line breaks to the beginning of the previous line. I think technically indented line breaks should only have 1 tab. perhaps you can rectify other instances of this in this PR? I know its confusing b/c I think we adopted slightly different standards throughout hera_cal development

i.e.
```
max_contiguous_flags: int or 2-list....
    to the largest...
```

Reply:
Unfortunately I don't know of a tool that will automatically reformat docstrings for you. However, once a consistent docstring format is adopted and adhered through throughtout the repo (e.g., `numpy`-style, which this is pretty close to), then you can have `flake8` check that they are formatted correctly. We do this on a few of our repos, here is how it's configured in pyuvdata: https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/setup.cfg. You just run `flake8` in the top-level of the repo and it'll check everything for you. (Flake8 does a lot more than docstrings, but you can pick and choose what checks to run.)

It also doesn't have to be an all-or-nothing thing, you can set it up so you only check certain files while the repo moves toward being compliant to a single style. (That's what pyuvdata does.) Also, for completeness, here's what this docstring should be in numpy style:
```
max_contiguous_flags : int or 2-list of ints
    The width of a region corresponding to the largest delay in
    the filter centers and filter_widths.
```
Basically, the return type is all that goes on the first line, and then the longer description is indented below. The full style guide is [here](https://numpydoc.readthedocs.io/en/latest/format.html).